File: c/src/hamming.c
#include "hamming.h"
#include <string.h>

int compute(const char* code1, const char* code2) {
    int diff = -1;
    int len1 = strlen(code1);
    int len2 = strlen(code2);
    if (len1 == len2) {
        diff = 0;
        for(int i = 0; i < len1; i++) {
            if (code1[i] != code2[i]) diff++;
        }
    }
    return diff;
}

File: fortran/src/hamming.f90
module hamming
  implicit none
contains

  function compute(strand1, strand2, distance)
      character(*) :: strand1, strand2
      integer :: distance, len1, len2, i
      logical :: compute

      distance = 0

      len1 = len(strand1)
      len2 = len(strand2)
      compute = len1 == len2
      if (compute) then
        do i = 1, len1
          if (strand1(i:i) /= strand2(i:i)) then
            distance = distance + 1
          end if
        end do
      end if

  end function compute

end module hamming
